Library/ViewModels/DiscoveryViewModel.swift

```diff
@@ -61,26 +65,42 @@ public protocol DiscoveryViewModelType {
   var outputs: DiscoveryViewModelOutputs { get }
 }
 
-private func initialParam() -> DiscoveryParams {
+public final class DiscoveryViewModel: DiscoveryViewModelType, DiscoveryViewModelInputs,
+DiscoveryViewModelOutputs {
+
+  private static func initialParams() -> DiscoveryParams {
+    let user = AppEnvironment.current.currentUser
+    let optedOutOfRecommendations = user?.optedOutOfRecommendations ?? true
 
+    if user == nil || optedOutOfRecommendations {
```

Comment:
I think the logic still confuses me a little with the `||` and the coalescing to `true`, how about this?

```swift
  private static func initialParams() -> DiscoveryParams {
    guard AppEnvironment.current.currentUser?.optedOutOfRecommendations == .some(false) else {
      return DiscoveryParams.defaults
        |> DiscoveryParams.lens.includePOTD .~ true
    }

    return DiscoveryParams.defaults
      |> DiscoveryParams.lens.includePOTD .~ true
      |> DiscoveryParams.lens.backed .~ false
      |> DiscoveryParams.lens.recommended .~ true
  }
```
The way I read this is - "By default we are opted out of recommendations (GDPR), so unless this is explicitly `false` (user intervention), we only display POTD." Does that sound right?

Reply:
I like this solution. It's much simpler!